audio_service_web/lib/audio_service_web.dart

```diff
@@ -19,130 +19,115 @@ class AudioServiceWeb extends AudioServicePlatform {
-              'seekforward',
-              () => handlerCallbacks?.fastForward(const FastForwardRequest()),
-            );
-            break;
-          case MediaActionMessage.stop:
-            session.setActionHandler(
-              'stop',
-              () => handlerCallbacks?.stop(const StopRequest()),
-            );
-            break;
-          default:
-            // no-op
-            break;
-        }
-      } catch (e) {
-        // TODO: handle this somehow?
+      switch (control.action) {
+        case MediaActionMessage.play:
+          session.setActionHandler(
+            'play',
+            () => handlerCallbacks?.play(const PlayRequest()),
+          );
+          break;
+        case MediaActionMessage.pause:
+          session.setActionHandler(
+            'pause',
+            () => handlerCallbacks?.pause(const PauseRequest()),
+          );
+          break;
+        case MediaActionMessage.skipToPrevious:
+          session.setActionHandler(
+            'previoustrack',
+            () =>
+                handlerCallbacks?.skipToPrevious(const SkipToPreviousRequest()),
+          );
+          break;
+        case MediaActionMessage.skipToNext:
+          session.setActionHandler(
+            'nexttrack',
+            () => handlerCallbacks?.skipToNext(const SkipToNextRequest()),
+          );
+          break;
+        // The naming convention here is a bit odd but seekbackward seems more
+        // analagous to rewind than seekBackward
+        case MediaActionMessage.rewind:
+          session.setActionHandler(
+            'seekbackward',
+            () => handlerCallbacks?.rewind(const RewindRequest()),
+          );
+          break;
+        case MediaActionMessage.fastForward:
+          session.setActionHandler(
+            'seekforward',
+            () => handlerCallbacks?.fastForward(const FastForwardRequest()),
+          );
+          break;
+        case MediaActionMessage.stop:
+          session.setActionHandler(
+            'stop',
+            () => handlerCallbacks?.stop(const StopRequest()),
+          );
+          break;
+        default:
+          // no-op
+          break;
       }
+
       for (final message in request.state.systemActions) {
         switch (message) {
           case MediaActionMessage.seek:
-            try {
-              setActionHandler('seekto', js.allowInterop((ActionResult ev) {
-                // Chrome uses seconds for whatever reason
-                handlerCallbacks?.seek(SeekRequest(
-                    position: Duration(
-                  milliseconds: (ev.seekTime * 1000).round(),
-                )));
-              }));
-            } catch (e) {
-              // TODO: handle this somehow?
-            }
+            setActionHandler('seekto', js.allowInterop((ActionResult event) {
+              handlerCallbacks?.seek(SeekRequest(
+                position: Duration(seconds: event.seekTime.toInt()),
+              ));
+            }));
```

Comment:
@nt4f04uNd You are dropping accuracy here... The `seekTime` is a double of seconds with fractions of a second. Calling `toInt` here will lead to a slightly strange/inaccurate seeking experience. Looks like you are doing something similar in the position state as well.

Reply:
Ahh, sure, thanks, I missed that. Fixed it.